config/v1alpha1/types_insights.go

```diff
@@ -48,24 +47,78 @@ type GatherConfig struct {
 	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The format for the disabledGatherer should be: {gatherer}/{function} where the function is optional.
+	// Gatherer consists of a lowercase string that may include underscores (_).
+	// Function consists of a lowercase string that may include underscores (_) and is separated from the gatherer by a forward slash (/).
 	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
 	// Run the following command to get the names of last active gatherers:
 	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
 	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	// +kubebuilder:validation:MaxItems=100
+	// +kubebuilder:validation:items:MaxLength=256
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, x.matches("^[a-z]+[_a-z]*[a-z]([/a-z][_a-z]*)?[a-z]$"))`,message=`disabledGatherers must be in the format of {gatherer}/{function} where the gatherer and function are lowercase strings that may include underscores (_) and are separated by a forward slash (/) if the function is provided`
 	// +optional
 	DisabledGatherers []string `json:"disabledGatherers"`
+	// storageSpec is an optional field that allows user to define persistent storage for on-demand gathering
+	// jobs to store the Insights data archive.
+	// If omitted, the gathering job will use ephemeral storage.
+	// +optional
+	StorageSpec *StorageSpec `json:"storageSpec,omitempty"`
+}
+
+// storageSpec provides persistent storage configuration options for on-demand gathering jobs.
+type StorageSpec struct {
+	// type is a required field that specifies the type of storage that will be used to store the Insights data archive.
+	// Valid value is "PersistentVolumeClaim".
+	// +required
+	Type StorageType `json:"type"`
+	// persistentVolume is an optional field that specifies the PersistentVolume that will be used to store the Insights data archive.
+	// The PersistentVolume must be created in the openshift-insights namespace.
+	// +optional
+	PersistentVolume PersistentVolumeConfig `json:"persistentVolume,omitempty"`
+}
+
+// storageType declares valid storage types
+// +kubebuilder:validation:Enum=PersistentVolumeClaim
+type StorageType string
+
+// persistentVolumeConfig provides configuration options for PersistentVolume storage.
+type PersistentVolumeConfig struct {
+	// persistentVolumeClaim is an optional field that specifies the configuration of the PersistentVolumeClaim that will
+	// be used to store the Insights data archive. The PersistentVolumeClaim must be created in the openshift-insights namespace.
+	// +optional
+	PersistentVolumeClaim PersistentVolumeClaimReference `json:"persistentVolumeClaim,omitempty"`
```

Comment:
Do you actually want this to be optional? With this field optional a user could do something like:
```yaml
storageSpec:
  type: PersistentVolumeClaim
  persistentVolume: {}
```

This _seems_ like it would be an invalid configuration based on my current understanding of your desired end state. 

If this is something you would like a user to be able to do, you should probably have some sort of default behavior that is documented.

Reply:
I think the main idea behind making it optional is to simplify adding more types in the future. For example, if I were to add back the `Ephemeral` type, I wouldn't want to define the `persistentVolume` as well. However, I'm not sure if it's possible to enforce a rule where, if the type is `PersistentVolumeClaim`, then `persistentVolume` must be defined.

If the type is `PersistentVolumeClaim` and `persistentVolume` is not defined we will default to ephemeral internally and log an error message for that

Reply:
Yes, it should be required